wolt/details.go

```diff
@@ -37,6 +37,30 @@ func (p *Participant) Name() string {
 }
 
 type Status string
+type DeliveryStatus string
+type DeliveryStatusToTimeMap map[DeliveryStatus]time.Time
+
+func (deliveryStatusLog *DeliveryStatusToTimeMap) UnmarshalJSON(bytes []byte) error {
+	o := &([]struct {
+		Datetime struct {
+			DateUnix int64 `json:"$date"`
+		} `json:"datetime"`
+		Status DeliveryStatus `json:"status"`
+	}{})
+
+	err := json.Unmarshal(bytes, o)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling DeliveryStatusToTimeMap %w", err)
+	}
+
+	res := make(DeliveryStatusToTimeMap)
+	for _, entry := range *o {
```

Comment:
Is it possible that the log will hold 2  entries for the same status? if so, the latest time should be added to the map

Reply:
I'm guessing that won't happen, but just in case I've made the map contain the latest timestamp.